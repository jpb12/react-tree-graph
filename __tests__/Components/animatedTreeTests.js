import React from 'react';
import { shallow } from 'enzyme';

import AnimatedTree from '../../src/components/animatedTree';

describe('<AnimatedTree>', () => {
	test('renders correctly', () => {
		const props = {
			data: {
				name: 'Colour',
				children: [{
					name: 'Black'
				}]
			},
			height: 100,
			width: 200
		};

		const tree = shallow(<AnimatedTree {...props}/>);

		expect(tree).toMatchSnapshot();
	});
});