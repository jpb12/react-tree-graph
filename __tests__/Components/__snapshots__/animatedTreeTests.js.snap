// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`<AnimatedTree> renders correctly 1`] = `
<Animated
  direction="ltr"
  duration={500}
  easing={[Function]}
  gProps={
    {
      "className": "node",
    }
  }
  getChildren={[Function]}
  height={100}
  keyProp="name"
  labelProp="name"
  links={
    [
      {
        "source": {
          "children": [
            Node {
              "data": {
                "name": "Black",
              },
              "depth": 1,
              "height": 0,
              "parent": Node {
                "children": [Circular],
                "data": {
                  "children": [
                    {
                      "name": "Black",
                    },
                  ],
                  "name": "Colour",
                },
                "depth": 0,
                "height": 1,
                "parent": null,
                "x": 40,
                "y": 0,
              },
              "x": 40,
              "y": 30,
            },
          ],
          "data": {
            "children": [
              {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 0,
          "y": 40,
        },
        "target": {
          "data": {
            "name": "Black",
          },
          "depth": 1,
          "height": 0,
          "parent": Node {
            "children": [
              Node {
                "data": {
                  "name": "Black",
                },
                "depth": 1,
                "height": 0,
                "parent": [Circular],
                "x": 40,
                "y": 30,
              },
            ],
            "data": {
              "children": [
                {
                  "name": "Black",
                },
              ],
              "name": "Colour",
            },
            "depth": 0,
            "height": 1,
            "parent": null,
            "x": 40,
            "y": 0,
          },
          "x": 30,
          "y": 40,
        },
      },
    ]
  }
  margins={
    {
      "bottom": 10,
      "left": 20,
      "right": 150,
      "top": 10,
    }
  }
  nodeProps={{}}
  nodeShape="circle"
  nodes={
    [
      {
        "children": [
          Node {
            "data": {
              "name": "Black",
            },
            "depth": 1,
            "height": 0,
            "parent": Node {
              "children": [Circular],
              "data": {
                "children": [
                  {
                    "name": "Black",
                  },
                ],
                "name": "Colour",
              },
              "depth": 0,
              "height": 1,
              "parent": null,
              "x": 40,
              "y": 0,
            },
            "x": 40,
            "y": 30,
          },
        ],
        "data": {
          "children": [
            {
              "name": "Black",
            },
          ],
          "name": "Colour",
        },
        "depth": 0,
        "height": 1,
        "parent": null,
        "x": 0,
        "y": 40,
      },
      {
        "data": {
          "name": "Black",
        },
        "depth": 1,
        "height": 0,
        "parent": Node {
          "children": [
            Node {
              "data": {
                "name": "Black",
              },
              "depth": 1,
              "height": 0,
              "parent": [Circular],
              "x": 40,
              "y": 30,
            },
          ],
          "data": {
            "children": [
              {
                "name": "Black",
              },
            ],
            "name": "Colour",
          },
          "depth": 0,
          "height": 1,
          "parent": null,
          "x": 40,
          "y": 0,
        },
        "x": 30,
        "y": 40,
      },
    ]
  }
  pathProps={
    {
      "className": "link",
    }
  }
  steps={20}
  svgProps={{}}
  textProps={{}}
  width={200}
/>
`;
